src/managers/KeybindManager.cpp

```diff
@@ -304,10 +304,10 @@ bool CKeybindManager::onKeyEvent(wlr_keyboard_key_event* e, SKeyboard* pKeyboard
 
     const auto         KEYCODE = e->keycode + 8; // Because to xkbcommon it's +8 from libinput
 
-    const xkb_keysym_t keysym         = xkb_state_key_get_one_sym(m_pXKBTranslationState, KEYCODE);
-    const xkb_keysym_t internalKeysym = xkb_state_key_get_one_sym(wlr_keyboard_from_input_device(pKeyboard->keyboard)->xkb_state, KEYCODE);
+    const xkb_keysym_t keysymByKeyboard = xkb_state_key_get_one_sym(wlr_keyboard_from_input_device(pKeyboard->keyboard)->xkb_state, KEYCODE);
```

Comment:
you can't do that I believe. We had an issue with this back in the day. Some mods alter the syms (eg. SHIFT+1 will now not work, because SHIFT and 1 produces a ! and not 1)


You need to make a copy of the state, initialize it with the keymap and not update the mods.

Reply:
Yeah, you're right. I did think about that at some point while reading xkb docs, but then promptly forgot again when I wrote the code. *puts on dunce hat*

Reply:
I resolved this by creating the equivalent of `m_pXKBTranslationState` for every keyboard whenever a keyboard's layout is changed. This way I don't need to copy it for every key event.